wrench-service/src/main/java/com/izettle/wrench/service/WrenchService.kt

```diff
@@ -0,0 +1,90 @@
+package com.izettle.wrench.service
+
+import android.app.IntentService
+import android.content.ContentResolver
+import android.content.Intent
+import android.net.Uri
+import com.izettle.wrench.core.Bolt
+import com.izettle.wrench.core.WrenchProviderContract
+
+class WrenchService : IntentService("WrenchService") {
+
+    override fun onHandleIntent(intent: Intent?) {
+        if (intent != null) {
+            val extras = intent.extras
+            for (key in extras!!.keySet()) {
+                val value = extras.get(key)
+                if (value is Int) {
+                    updateInteger(contentResolver, key, value)
+                } else if (value is String) {
+                    updateString(contentResolver, key, value)
+                } else if (value is Boolean) {
+                    updateBoolean(contentResolver, key, value)
+                }
+            }
+        }
+    }
+
+    private fun updateInteger(contentResolver: ContentResolver, key: String, value: Int) {
+        var bolt = getBolt(contentResolver, key) ?: return
+
+        if (bolt.id == 0L) {
+            bolt = bolt.copy(bolt.id, Int::class.java.name, key, value.toString())
+            val uri = insertBolt(contentResolver, bolt)
+            bolt.id = java.lang.Long.parseLong(uri!!.lastPathSegment)
```

Comment:
`bolt.id = uri?.lastPathSegment?.toLong() ?: throw IllegalArgumentException("URI should not be null")`